src/Components/GEOSldas_GridComp/GEOSmetforce_GridComp/LDAS_Forcing.F90

```diff
@@ -5540,6 +5362,84 @@ subroutine GEOS_openfile(FileOpenedHash, fname_full, fid, tile_coord, m_hinterp,
     
   end subroutine GEOS_openfile
 
+  subroutine get_neighbor_index(m_hinterp, tile_coord, ll_lon, ll_lat, dlon, dlat, N_lon, N_lat, i1, j1, i2, j2, x1, y1, x2, y2, rc) 
+    ! compute indices of nearest neighbors needed for bilinear
+    ! interpolation from GEOSgcm grid to tile space
+    integer, intent(in) :: m_hinterp
+    type(tile_coord_type), dimension(:), intent(in)    :: tile_coord
+    real, intent(in) :: ll_lat, ll_lon, dlat, dlon
+    integer, intent(in) :: N_lon, N_lat
+    integer, dimension(:), intent(inout) :: i1, j1
+    integer, dimension(:), intent(inout) :: i2, j2
+    real,    dimension(:), intent(out) :: x1, x2, y1, y2
+    integer, optional, intent(out) :: rc
+
+    real, dimension(:), allocatable :: xnew, ynew, tmp_lat, tmp_lon
+    integer, dimension(:), allocatable :: inew, jnew
+    character(len=*), parameter :: Iam="get_bilinear_neighbors" 
+    real  :: offset
+
+    ! ll_lon and ll_lat refer to lower left corner of grid cell
+    ! (as opposed to the grid point in the center of the grid cell)
+
+    ! pchakrab: For bilinear interpolation, for each tile, we need:
+    !  x1, x2, y1, y2 (defining the co-ords of four neighbors) and
+    !  i1, i2, j1, j2 (defining the indices of four neighbors)
+    offset = 0.0001
+    ! add small offsets to avoid unpredictable assignment of
+    ! regularly spaced tiles (such as from the EASEv2 tile space)
+    ! to ERA5 grid cells along certain lat/lon values
+    ! (that is, make it possible for post-processing scripts in other
+    !  languages to exactly reproduce the mapping that is done here)
+    ! TO DO: add if statement so the offset is only applied when
+    !        the model is run in the EASE grid tile space
+    i1 = ceiling((tile_coord%com_lon + offset - ll_lon)/dlon)
+    j1 = ceiling((tile_coord%com_lat + offset - ll_lat)/dlat)
+    ! NOTE: For a "date line on center" grid and (180-dlon/2) < lon < 180
+    !  we now have i1=(grid%N_lon+1)
+    ! This needs to be fixed as follows:
+    where( i1 > N_lon)  i1=1
+    if (any(j1 > N_lat)) then
+       call ldas_abort(LDAS_GENERIC_ERROR, Iam, "encountered tile near the poles")
+    end if
+
+    if (m_hinterp /= 1) then
+      RETURN_(ESMF_SUCCESS)
+    endif
+
+    x1 = (i1-1)*dlon + ll_lon  + dlon/2.
+    y1 = (j1-1)*dlat + ll_lat  + dlat/2.
+    
+    tmp_lon = tile_coord%com_lon + dlon/2.
+    tmp_lat = tile_coord%com_lat + dlat/2.
+   
+    inew =  ceiling((tmp_lon  - ll_lon)/dlon)   ! nearest-neighbor i index of shifted location
+    jnew =  ceiling((tmp_lat  - ll_lat)/dlat)   ! nearest-neighbor j index of shifted location
```

Comment:
I can't say for sure, but I think to be consistent we also need to apply "offset" in this calculation.
@biljanaorescanin's branch for ERA5_LIS bilinear interpolation also doesn't  do this consistently.
Perhaps that's why experiments with M-2 forcing have roundoff diffs vs. this branch.  (Note that "develop" uses the same "this_lon" and "this_lat" in both "ceiling()" operations for determining (i) the nearest neighbor and (ii) the grid cell that sits diagonally across from the nearest neighbor).
Please add "+offset" into lines 5416 and 5417 and try the M2 experiment again. 

If this doesn't work, we'll need to keep looking.  I note that the implementation of get_neighbor_index() is somewhat different from the corresponding code in the original GEOS_OpenFile() on develop -- which is what we're comparing against when running with M2 forcing.  There may still be a difference in the placement of the "wrap-around" statements between get_neighbor_index() and GEOS_OpenFile() on develop.